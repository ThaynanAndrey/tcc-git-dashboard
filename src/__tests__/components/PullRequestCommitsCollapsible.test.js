import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestCommitsCollapsible from '../../components/project/collapsible/pullRequestDetails/PullRequestCommitsCollapsible';

describe('Test PullRequestCommitsCollapsible', () => {
  let tree;
  
  beforeEach(() => {
    const pullRequest = {};

    tree = shallow(
      <PullRequestCommitsCollapsible
        pullRequest={pullRequest}
      />
    );
  });

  it('renders correctly a component to show PullRequestCommitsCollapsible', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  
  it('the component should have class collapsible-header', () => {
    expect(tree.find(".collapsible-header").exists()).toBeTruthy();
  });

  it('the component should not have classes collapsible-body body-collapsible-project', () => {
    expect(tree.find(".collapsible-body.body-collapsible-project").exists()).toBeFalsy();
    expect(tree.find("table").exists()).toBeFalsy();
  });

  it('the component should have classes collapsible-body body-collapsible-project when pullRequest prop has commits', () => {
    const pullRequest = {
        commits: [{
            author: {name: "Teste"},
            date: "10/10/2010",
            message: "nada"
        }]
    };

    tree = shallow(
      <PullRequestCommitsCollapsible
        pullRequest={pullRequest}
      />
    );

    expect(tree.find(".collapsible-body.body-collapsible-project").exists()).toBeTruthy();
    expect(tree.find("table").exists()).toBeTruthy();
  });
})