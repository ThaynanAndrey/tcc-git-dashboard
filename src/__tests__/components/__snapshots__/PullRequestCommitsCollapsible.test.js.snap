// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestCommitsCollapsible renders correctly a component to show PullRequestCommitsCollapsible 1`] = `
<CollapsibleCard
  isActive={false}
>
  <div
    className="collapsible-header"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": "0.5rem",
            "color": "white",
            "height": "27px",
            "marginRight": "1rem",
            "textAlign": "center",
            "width": "2rem",
          }
        }
      />
      Commits
    </div>
  </div>
</CollapsibleCard>
`;
